### PR TITLE
Add diag_aug option to MASE and OMNI

### PR DESCRIPTION
**Pull request by j1c**
#### What does this implement/fix? Explain your changes.
- Adds diag_aug option to Omni and MASE. 
- In Omni, diag_aug occurs prior to omni matrix creation.
- In MASE, diag_aug occurs prior to the first individual embeddings.

### CODE REVIEW

File: graspy/embed/omni.py [hunk 1]
```diff
@@ -173,6 +180,14 @@ def fit(self, graphs, y=None):
                 )
                 warnings.warn(msg, UserWarning)
 
+        # Diag augment
```
Comment (bdpedigo) [inline graspy/embed/omni.py:183]: this looks identical to the MASE one? maybe we have a `_multigraph_augment_diagonal` method in `BaseEmbedMulti`?

File: tests/test_mase.py [hunk 2]
```diff
@@ -64,84 +68,81 @@ def test_bad_inputs():
...
 
+def test_diag_aug():
+    np.random.seed(5)
+    n = 100
+    p = 0.25
+
+    graphs_list = [er_np(n, p) for _ in range(2)]
+    graphs_arr = np.array(graphs_list)
+
+    omni_list = MultipleASE(diag_aug=True).fit_transform(graphs_list)
```
Comment (bdpedigo) [inline tests/test_mase.py:79]: change omni to mase just for clarity?

File: graspy/embed/mase.py [hunk 3]
```diff
@@ -26,16 +26,16 @@ class MultipleASE(BaseEmbedMulti):
...
+    allowed to vary for each graph, but are symmetric. All graphs share a common a
+    latent position matrix :math:`V`.
+
     For a population of directed graphs, MASE assumes that the population is sampled
-    from :math:`UR^{(i)}V^T` where :math:`U \in \mathbb{R}^{n\times d_1}`, 
-    :math:`V \in \mathbb{R}^{n\times d_2}`, and 
-    :math:`R^{(i)} \in \mathbb{R}^{d_1\times d_2}`. In this case, score matrices 
-    :math:`R^{(i)}` can be assymetric and non-square, but all graphs still share a 
+    from :math:`UR^{(i)}V^T` where :math:`U \in \mathbb{R}^{n\times d_1}`,
+    :math:`V \in \mathbb{R}^{n\times d_2}`, and
```
Comment (bdpedigo) [inline graspy/embed/mase.py:36]: is a lot of this diff redundant with #408 ?

Comment (bdpedigo) [reply graspy/embed/mase.py:36]: all the docstring changes i mean

File: graspy/embed/omni.py (hunk 1 above)
Comment (j1c) [reply graspy/embed/omni.py:183]: Makes sense. Done.

File: tests/test_mase.py (hunk 2 above)
Comment (j1c) [reply tests/test_mase.py]: Done.

File: graspy/embed/base.py [hunk 4]
```diff
@@ -243,3 +243,30 @@ def _check_input_graphs(self, graphs):
...
+            If input is list of ndarray, then list is returned.
+            If input is ndarray, then ndarray is returned.
+        """
+        if isinstance(graphs, list):
+            out = [augment_diagonal(g) for g in graphs]
+        elif isinstance(graphs, np.ndarray):
+            # Copying is necessary to not overwrite input array
+            out = np.array([augment_diagonal(graphs[i]) for i in range(self.n_graphs_)])
+
+        return graphs
```
Comment (bdpedigo) [inline graspy/embed/base.py:272]: return `out`?

**Review by bdpedigo — APPROVED** [review]
looks good to me - only other thing is it'd be nice to have a test showing that `diag_aug` actually does something, I guess for ASE also... but that test would probably require showing that an estimated latent distribution is closer to a true one with `diag_aug` on, and then doing that for multiple graphs... so meh I'm fine without it for now.

**Pull request merged**